plugins/modules/provision_workflow_manager.py

```diff
@@ -1003,31 +1076,63 @@ def reprovision_wired_device(self):
         """
+        if self.compare_dnac_versions(self.get_ccc_version(), "2.3.5.3") <= 0:
+            try:
+                response = self.dnac_apply['exec'](
+                    family="sda",
+                    function="re_provision_wired_device",
+                    op_modifies=True,
+                    params=self.want["prov_params"],
+                )
+                taskid = response.get("taskId")
+                while True:
+                    result = self.get_task_details(taskid)
+
+                    if "processcfs_complete=true" in result.get("data"):
+                        self.msg = ("Wired Device '{0}' re-provisioning completed successfully.".format(self.device_ip))
+                        self.log(self.msg, "INFO")
+                        self.result["changed"] = True
+                        self.result['msg'] = ("Wired Device '{0}' re-provisioning completed successfully.".format(self.device_ip))
+                        self.result['response'] = self.msg
+                        self.log(self.result['msg'], "INFO")
+                        return self
+
+                    elif result.get("isError") is True:
+                        raise Exception
+
+            except Exception as e:
+                self.msg = "Error in re-provisioning device '{0}' due to {1}".format(self.device_ip, str(e))
+                self.log(self.msg, "ERROR")
+                self.result['response'] = self.msg
+                self.status = "failed"
+                self.check_return_status()
+        else:
+            try:
+                response = self.dnac_apply['exec'](
+                    family="sda",
+                    function="re_provision_devices",
+                    op_modifies=True,
+                    params={"payload": reprovision_param}
+                )
 
-        try:
-            response = self.dnac_apply['exec'](
-                family="sda",
-                function="re_provision_wired_device",
-                op_modifies=True,
-                params=self.want["prov_params"],
-            )
-            self.log(self.want["prov_params"])
-            task_id = response.get("taskId")
-            self.get_task_status(task_id=task_id)
-            self.result["changed"] = True
-            self.result['msg'] = "Re-Provision for device '{0}' done successfully".format(self.device_ip)
-            self.result['diff'] = self.validated_config
-            self.result['response'] = task_id
-            self.log(self.result['msg'], "INFO")
-            return self
-        except Exception as e:
-            self.msg = "Error in re-provisioning device '{0}' due to {1}".format(self.device_ip, str(e))
-            self.log(self.msg, "ERROR")
-            self.result['response'] = self.msg
-            self.status = "failed"
-            self.check_return_status()
+                self.check_tasks_response_status(response, api_name='re_provision_devices')
```

Comment:
self.log("Task status after 're_provision_devices' execution: {0}".format(self.status), "DEBUG")

Reply:
Updated the code accordingly.